pkg/controllermanager/controller/plant/plant_control.go

```diff
@@ -326,11 +326,10 @@ func (c *defaultPlantControl) healthChecks(ctx context.Context, healthChecker *H
 	}()
 	go func() {
 		defer wg.Done()
-		newNodes, err := healthChecker.CheckPlantClusterNodes(ctx, &healthyNodes)
-		healthyNodes = newConditionOrError(healthyNodes, *newNodes, err)
+		conditionEveryNodeReady = *healthChecker.CheckPlantClusterNodes(ctx, healthChecker, conditionEveryNodeReady)
```

Comment:
Why dereference if `CheckPlantClusterNodes` anyways needs the pointer?
I'd also suggest to just call it `everyNodeReady` to align with the other health check conditions.